tests/multiple_hooks_modify_returnvalue.phpt

```diff
@@ -0,0 +1,17 @@
+--TEST--
+Check if hooks receive modified returnvalue
+--EXTENSIONS--
+otel_instrumentation
+--FILE--
+<?php
+\OpenTelemetry\Instrumentation\hook(null, 'helloWorld', post: fn(mixed $object, array $params, string $return): string => ++$return);
```

Comment:
This callback receives the original value (`"a"`) instead of the updated value (`"b"`), we update `execute_data` without updating `$returnValue`:
https://github.com/open-telemetry/opentelemetry-php-instrumentation/blob/f2c1a1e87ef540f2d70e2d3bdc990a32f04307ec/otel_observer.c#L257-L258